client.go

```diff
@@ -356,6 +356,115 @@ func (c *Client) logf(format string, v ...interface{}) {
 	}
 }
 
+// Broadcast a message to all members in a Group
+func (c *Client) Broadcast(timeout time.Duration, g *Group, msg interface{}) (BroadcastResult, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	return c.broadcast(ctx, cancel, g, msg)
+}
+
+// BroadcastC (broadcast) a message to all members in a Group. The context can be used to control
+// cancellations or timeouts
+func (c *Client) BroadcastC(ctx context.Context, g *Group, msg interface{}) (BroadcastResult, error) {
+	cont, cancel := context.WithCancel(ctx)
+	defer cancel()
+	return c.broadcast(cont, cancel, g, msg)
+}
+
+func (c *Client) broadcast(ctx context.Context, cancel context.CancelFunc, g *Group, msg interface{}) (BroadcastResult, error) {
+	res := make(BroadcastResult)
+	receivers := g.Members()
+
+	var broadcastErr error
+	mu := new(sync.Mutex)
+	wg := new(sync.WaitGroup)
+	for _, rec := range receivers {
+		wg.Add(1)
+		go func(receiver string) {
+			defer wg.Done()
+			resp, err := c.RequestC(ctx, receiver, msg)
+			switch {
+			case err == nil && g.fastest:
+				// if this request was successful and the group is configured to Fastest,
+				// then cancel the context so other requests are terminated
+				cancel()
+			case err != nil && !g.fastest:
```

Comment:
Youll need to save the error even if `g.fastest` is true.   otherwise what happens if all requests fail?   May I suggest that you set broadcastErr in all cases, but after `wg.Wait()` that you unset it if any of the request completed.